File: offer/Tencent/2.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

double compute(int l,int r,int k){
    double res = 0.0;
    for(int i=l;i<=r;i++){
        res += pow(i+pow(10,-k),1.0/3.0)-pow(i,1.0/3.0);
    }
    return res;
}

int main(){
    int T;
    cin>>T;
    vector<double> res;
    int l,r,k;
    for(int i=0;i<T;i++){
        cin>>l>>r>>k;
        res.push_back(compute(l,r,k));
    }
    for(int i=0;i<T;i++){
        cout<<res[i]<<endl;
    }
    return 0;
}

File: offer/Tencent/2.py
import math
def compute(l,r,k):
    res = 0
    for i in range(l,r+1):
        res += math.pow(i+math.pow(10,-k),1/3)-math.pow(i,1/3)
    return res
n = int(input())
res = []
for i in range(n):
    s = input().split(" ")
    res.append(compute(int(s[0]),int(s[1]),int(s[2])))

for i in res:
    print(i)